### ISSUE TITLE
Support for updating parent section of pom

### ISSUE THREAD

**Issue by ryandawsonuk** [jenkins-x/updatebot#36]
Looking at https://github.com/jenkins-x/updatebot/blob/bb7459be5ba443bce9f9c71a968143725d45313e/updatebot-core/src/main/java/io/jenkins/updatebot/kind/maven/PomHelper.java it seems the updatebot can currently update the dependencies, plugins and properties of a downstream project's pom.xml file. My impression is that it doesn't update a parent section of a pom file.

We are building https://github.com/ryandawsonuk/activiti-build with the Jenkinsfile and .updatebot.yml there. The updatebot runs without error and logs `io.jenkins.updatebot.commands.PushSourceChanges - push source changes from . to ryandawsonuk/Activiti changes found: org.activiti:activiti-parent => 7.0.6` but https://github.com/ryandawsonuk/Activiti/blob/develop/pom.xml doesn't get updated. I am guessing this is because the artifact is used in the parent https://github.com/ryandawsonuk/Activiti/blob/develop/pom.xml#L5

It would also be good to be able to push changes to the dependencyManagement section of a pom. It's not clear to me whether that would be updated or not.

**Comment by ryandawsonuk** [comment]
We've created a simpler reproducer now. There's an ['upstream' project](https://github.com/ryandawsonuk/upstream) which has the [.updatebot.yaml ](https://github.com/ryandawsonuk/upstream/blob/master/.updatebot.yml)and a [Jenkinsfile that invokes the updatebot](https://github.com/ryandawsonuk/upstream/blob/358540cad30a454a179e5506a8ef6acd802cc041/Jenkinsfile#L68).  When the pipeline runs it pushes to the ['downstream' project](https://github.com/ryandawsonuk/downstream) and it does [generate a PR ](https://github.com/ryandawsonuk/downstream/pull/1)to update the dependency in the <dependencies> section but not the parent section of the pom.

(We've also noticed that `updatebot push` doesn't do push anything but "updatebot push-version --kind maven org.example:upstream \$(cat VERSION)" does. This bit isn't a big problem for us though.)

It's also worth noting that it took us a while to figure out that updatebot was already available to use in the maven build pods and that we didn't need to install the jenkins plugin so it might be worth updating the documentation somewhere to say that.

**Comment by ryandawsonuk** [comment]
We created a simpler reproducer as well. There's an 'upstream' project which has the .updatebot.yaml and a Jenkinsfile that invokes the updatebot. When the pipeline ran using current updatebot it pushed to the 'downstream' project and it did generate a PR to update the dependency in the section but not the parent section of the pom. We then built a version of updatebot using the pr and ran that to see the parent updated. 

We also noticed that updatebot push didn't detect changes but "updatebot push-version --kind maven org.example:upstream $(cat VERSION)" did push. This turned out to be because the update it points at and invokes a version of updatebot Maven plugin and it was pointing at an old fabric8 one.

It's also worth noting that it took us a while to figure out that updatebot was already available to use in the maven build pods and that we didn't need to install the jenkins plugin so it might be worth updating the documentation somewhere to say that.
